polar-language-server/src/lib.rs

```diff
@@ -264,45 +272,74 @@ impl PolarLanguageServer {
             .collect()
     }
 
-    fn document_from_polar_error_context(&self, e: &PolarError) -> Option<TextDocumentItem> {
-        uri_from_polar_error_context(e).and_then(|uri| {
+    fn document_from_polar_diagnostic_context(
+        &self,
+        d: &PolarDiagnostic,
+    ) -> Option<TextDocumentItem> {
+        uri_from_polar_error_context(d).and_then(|uri| {
             if let Some(document) = self.documents.get(&uri) {
                 Some(document.clone())
             } else {
                 let tracked_docs = self.documents.keys().map(ToString::to_string);
                 let tracked_docs = tracked_docs.collect::<Vec<_>>().join(", ");
                 log(&format!(
                     "untracked doc: {}\n\tTracked: {}\n\tError: {}",
-                    uri, tracked_docs, e
+                    uri, tracked_docs, d
                 ));
                 None
             }
         })
     }
 
-    /// Create a `Diagnostic` from a `PolarError`, filtering out "ignored" errors.
-    fn diagnostic_from_polar_error(
+    /// Create `Diagnostic` from `polar_core::diagnostic::Diagnostic`, filtering out "ignored"
+    /// diagnostics.
+    fn diagnostic_from_polar_diagnostic(
         &self,
-        e: &PolarError,
-    ) -> Option<(TextDocumentItem, Diagnostic)> {
+        d: PolarDiagnostic,
```

Comment:
function is a bit long to use use `d` throughout and you also have `docs` maybe make this the full name

Reply:
29b8207b